IMAGE:
  WIDTH: 640
  HEIGHT: 480
  ROI_START_HEIGHT: 350
  ROI_HEIGHT: 50

XYCAR:
  START_SPEED: 0.0
  MAX_SPEED: 5.0
  MIN_SPEED: 5.0
  SPEED_CONTROL_THRESHOLD: 15.0
  ROTATE_THRESHOLD: 10
  ACCELERATION_STEP: 0.05
  DECELERATION_STEP: 0.1
  LINEAR_UNIT: 10
  ANGLE_UNIT: 1

PID:
  P_GAIN: 0.36
  I_GAIN: 0.0006
  D_GAIN: 0.08

STANLEY:
  K_GAIN: 0.032
  LOOK_AHREAD_DISTANCE: 0.2

STOP:
  SAMPLE_SIZE: 25
  PROBABILITY: 0.9

MOVING_AVERAGE_FILTER:
  SAMPLE_SIZE: 30

TOPIC:
  PUB_NAME: /xycar_motor
  SUB_NAME: /usb_cam/image_raw/
  QUEUE_SIZE: 1

CANNY:
  LOW_THRESHOLD: 150
  HIGH_THRESHOLD: 250

DETECTION:
  CLASSES:
    [
      "LEFT",
      "RIGHT",
      "STOP",
      "CROSSWLAK",
      "UTURN",
      "TRAFFIC_LIGHT",
      "XYCAR",
      "IGNORE",
    ]
  CONF_THRESHOLD: 0.7
  SIGN_LIFESECOND: 0.5
  MIN_BOUNDINGBOX_AREA: 2000

LIDAR:
  RANGE_RADIAN_THRESHOLD: 1.0681
  RANGE_DISTANCE_THRESHOLD: 0.35
  CLUSTER_DISTANCE_THRESHOLD: 0.04
  CLUSTER_MIN_POINT: 3
  CLUSTER_MAX_POINT: 50
  TRACKING_DISTANCE_THRESHOLD: 0.15
  TRACKING_MISS_SECOND: 1


AVOIDANCE_INPUT:
  POSITION: 0.05
  SLOPE: 0.0001

ROTATE_INPUT:
  POSITION: 0.125
  SLOPE: 0.0001

SIGN_INPUT:
  POSITION: 10
  SLOPE: 0.0001

HOUGH:
  ABS_SLOPE_RANGE: 10.0
  THRESHOLD: 15
  MIN_LINE_LENGTH: 15
  MAX_LINE_GAP: 10
  CONTINUS_THRESHOLD: 200
  CLUSTER_DISTANCE: 40
  CLUSTER_RADIAN: 0.174533

DEBUG: true
